### ISSUE TITLE
Natural sounding turn-by-turn directions

### ISSUE THREAD

**Issue by robintown** [osmandapp/Osmand#9063]
Currently, turn-by-turn directions sound very unnatural (in English and many other languages), since the app uses only a few strings for the many different situations encountered during navigation. I'm referring to the strings with IDs that begin with "route_".

A few notable issues:
- The strings have no placeholders for road names or other information (which isn't great for localization).
- The strings make no distinction between turning onto a named road and turning onto an unnamed road. This results in awkwardly worded directions, like, "Head Galaxie Road", "Turn left and go Impala Drive", "Head", and "Turn slightly right and go".
- It would also be useful to make the distinction between keeping right to stay on the same road, versus keeping right to turn onto a new road, as has been suggested for voice navigation in #8329.

This problem already seems to have been solved for voice navigation, so is there any way that the app could reuse that logic for producing turn-by-turn directions?

**Comment by sonora** [comment]
Yes, the turn instructions so far are produced as a very simplistic 'one size fits languages' notebook style structure.

It would be a nightmare to maintain a second system of correct grammar, semantics,  and complex vocabulary for several dozen languages like we do for the TTS voice prompts. And to what extent we could re-use the structure already maintained for voice prompts would need some research, definitely not a low-effort task.

**Comment by sonora** [comment]
Plus: The output is used as captions in the notifications, meaning a certain telegram style brevity is actually essential. Not actually sure there is much value-add or return of invest for big changes here...

**Comment by frasty** [comment]
I chime in just to suggest that it would very beneficial (if it is feasible) to disable turn instructions for turns that are actually bends of the road and that are not heading onto other roads. This would decrease the amount of useless turn instructions especially for routing across mountain roads.
